<?xml version="1.0"?>
<launch>
  <!-- AR Marker Laser Correction Launch File -->
  
  <!-- Arguments -->
  <arg name="base_frame" default="base_link" />
  <arg name="laser_topic" default="/scan" />
  <arg name="ar_marker_topic" default="/ar_pose_marker" />
  <arg name="search_radius" default="0.2" />
  <arg name="angle_tolerance" default="45.0" />
  <arg name="min_line_length" default="0.05" />
  <arg name="publish_visualization" default="true" />
  
  <!-- Laser Line Extraction -->
  <node pkg="laser_line_extraction" type="laser_line_extraction_node" 
        name="laser_line_extraction" output="screen">
    <remap from="scan" to="$(arg laser_topic)" />
    <rosparam command="load" file="$(find ar_marker_laser_correction)/config/line_extraction_params.yaml" />
  </node>
  
  <!-- AR Marker Laser Correction Node -->
  <node pkg="ar_marker_laser_correction" type="ar_marker_laser_correction_node" 
        name="ar_marker_laser_correction" output="screen">
    <remap from="ar_pose_marker" to="$(arg ar_marker_topic)" />
    <param name="base_frame" value="$(arg base_frame)" />
    <param name="search_radius" value="$(arg search_radius)" />
    <param name="angle_tolerance" value="$(arg angle_tolerance)" />
    <param name="min_line_length" value="$(arg min_line_length)" />
    <param name="publish_visualization" value="$(arg publish_visualization)" />
  </node>
  
  <!-- RViz for visualization (optional) -->
  <node if="$(arg publish_visualization)" pkg="rviz" type="rviz" name="rviz" 
        args="-d $(find ar_marker_laser_correction)/config/correction_viz.rviz" 
        required="false" />

</launch>
